quipucords/scanner/network/runner/roles/redhat_release/tasks/main.yml

```diff
@@ -4,25 +4,17 @@
   set_fact:
     internal_host_started_processing_role: "redhat_release"
 
-- name: initialize redhat_release facts
-  set_fact:
-    redhat_release_name: ''
-    redhat_release_version: ''
-    redhat_release_release: ''
-  ignore_errors: yes
-
 - name: gather internal_redhat_release fact
   raw: rpm -q --queryformat "%{NAME}\n%{VERSION}\n%{RELEASE}\n" --whatprovides redhat-release
   register: internal_redhat_release
-  ignore_errors: yes
+  ignore_errors: true
   when: 'internal_have_rpm'
 
 - name: set redhat_release facts based on internal_redhat_release
   set_fact:
     redhat_release_name: "{{ internal_redhat_release['stdout_lines'][0] }}"
     redhat_release_version: "{{ internal_redhat_release['stdout_lines'][1] }}"
     redhat_release_release: "{{ internal_redhat_release['stdout_lines'][2] }}"
-  ignore_errors: yes
+  ignore_errors: true
   when:
     - "'stdout_lines' in internal_redhat_release"
-    - "internal_redhat_release['stdout_lines']|length == 3"
```

Comment:
Why the length would be longer than 3? Probably if it raised an error, therefore o failed_when would be much more efficient than ignore_errors. 
I guess this calls for a refactor. hehe =P

Reply:
It's called "defensive programming". If the output don't have 3 lines, something is wrong.